proxy/http2/Http2Stream.cc

```diff
@@ -356,8 +356,6 @@ Http2Stream::transaction_done()
 
   if (closed) {
     // Safe to initiate SSN_CLOSE if this is the last stream
-    if (cross_thread_event)
-      cross_thread_event->cancel();
     // Schedule the destroy to occur after we unwind here.  IF we call directly, may delete with reference on the stack.
```

Comment:
Sorry, I agree that the title and description is misleading, I meant to say cancelling, not closing. Modifying title and description to be more correct.